lib/rucio/daemons/storage/consistency/actions.py

```diff
@@ -0,0 +1,771 @@
+        print("no scans available for this RSE")
+
+
+
+
+
+
+
+def deckard_loop(rses, dark_min_age, dark_threshold_percent, miss_threshold_percent,\
+  force_proceed, scanner_files_path):
+    print("\n A loop over all RSEs")
+    for rse in rses:
+        print("Now processing:", rse)
+        deckard(rse, dark_min_age, dark_threshold_percent, miss_threshold_percent,\
+          force_proceed, scanner_files_path)
+
+def actions_loop(once, rses, sleep_time, dark_min_age, dark_threshold_percent,\
+  miss_threshold_percent, force_proceed, scanner_files_path):
+
+    """
+    Main loop to apply the CC actions
+    """
+    hostname = socket.gethostname()
+    pid = os.getpid()
+    current_thread = threading.current_thread()
+    print("hostname:", hostname, " pid:", pid, " current_thread:", current_thread)
+
+    # Make an initial heartbeat
+    # so that all storage-consistency-actions have the correct worker number on the next try
+
+    executable = 'storage-consistency-actions'
+    heartbeat = live(executable=executable, hostname=hostname, pid=pid, thread=current_thread)
+    prefix = 'storage-consistency-actions[%i/%i] ' % (heartbeat['assign_thread'],\
+      heartbeat['nr_threads'])
+    logger = formatted_logger(logging.log, prefix + '%s')
+    graceful_stop.wait(1)
+
+    while not graceful_stop.is_set():
+        try:
+            # heartbeat
+            heartbeat = live(executable=executable, hostname=hostname, pid=pid,\
+              thread=current_thread)
+            print("\nheartbeat?", heartbeat)
+            prefix = 'storage-consistency-actions[%i/%i] ' % (heartbeat['assign_thread'],\
+              heartbeat['nr_threads'])
+            print("\nprefix:", prefix)
+            logger = formatted_logger(logging.log, prefix + '%s')
+            start = time.time()
+            print("\nStartTime:", start)
+            logger(logging.DEBUG, 'fake query time %f' % (time.time() - start))
+
+            deckard_loop(rses, dark_min_age, dark_threshold_percent, miss_threshold_percent,\
+              force_proceed, scanner_files_path)
+            daemon_sleep(start_time=start, sleep_time=sleep_time, graceful_stop=graceful_stop,\
+              logger=logger)
+
+
+        except Exception as e:
+            traceback.print_exc()
+            print("\nSomething went wrong here...", e)
+            print("\nSomething went wrong here...", e.__class__.__name__)
+        if once:
+            break
+
+    die(executable=executable, hostname=hostname, pid=pid, thread=current_thread)
+
+
+def stop(signum=None, frame=None):
+    """
+    Graceful exit.
+    """
+    graceful_stop.set()
+
+
+def run(once=False, rses=None, sleep_time=60, dark_min_age=28, dark_threshold_percent=1.0,\
+  miss_threshold_percent=1.0, force_proceed=False, scanner_files_path="/var/cache/consistency-dump",\
+  threads=1):
+    """
+    Starts up the Consistency-Actions.
+    """
+    print("\nNow inside run(once...)\n")
+    if rses == []:
```

Comment:
Take a look at the reaper code. It allows RSEs to be specified as an RSE expression.

Reply:
The purpose of this statement is just to handle the case of daemon operating normally, vs operating in a single-RSE mode (interactively). So, no more cases need to be distinguished here. I believe it's OK as it is.